### PR TITLE
add tfnet example

### PR DESCRIPTION
**Pull request by yangw1234**

### CODE REVIEW

**Comment by yangw1234** [comment]
the example tested on spark local and a cluster for correctness

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/net/TFNet.scala [hunk 1]
```diff
@@ -55,15 +54,11 @@ class TFNet(graphDef: GraphDef,
   }
```
Comment (jason-dai) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/net/TFNet.scala]: change to `if (output.isTensor)`

File: zoo/src/main/scala/com/intel/analytics/zoo/examples/tfnet/Predict.scala [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+  case class PredictParam(
+    image: String = "/tmp/datasets/cat_dog/train_sampled",
+    outputFolder: String = "data/demo",
+    model: String = "/tmp/models/ssd_mobilenet_v1_coco_2017_11_17" +
+      "/frozen_inference_graph.pb",
+    classNamePath: String = "/tmp/models/coco_classname.txt",
+    nPartition: Int = 4)
+
+  val parser = new OptionParser[PredictParam]("Analytics Zoo Object Detection Demo") {
+    head("Analytics Zoo Object Detection Demo")
```
Comment (jason-dai) [inline zoo/src/main/scala/com/intel/analytics/zoo/examples/tfnet/Predict.scala]: TFNet Object Detection Demo

File: zoo/src/main/scala/com/intel/analytics/zoo/examples/tfnet/Predict.scala [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    opt[Int]('p', "partition")
+      .text("number of partitions")
+      .action((x, c) => c.copy(nPartition = x))
+  }
+
+  def main(args: Array[String]): Unit = {
+    parser.parse(args, PredictParam()).foreach { params =>
+      val conf = new SparkConf()
+        .setAppName("Object Detection Example")
+      val sc = NNContext.getNNContext(conf)
```
Comment (jason-dai) [inline zoo/src/main/scala/com/intel/analytics/zoo/examples/tfnet/Predict.scala]: change to `NNContext.getNNContext("TFNet Object Detection Example")`

File: zoo/src/main/scala/com/intel/analytics/zoo/examples/tfnet/Predict.scala [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+      val model = Sequential()
+      model.add(Transpose(Array((2, 4), (2, 3))))
+      model.add(Contiguous())
+      model.add(detector)
+
+      val data = ImageSet.read(params.image, sc, minPartitions = params.nPartition)
+        .transform(ImageResize(256, 256) -> ImageMatToTensor() -> ImageSetToSample())
+      val output = model.predictImage(data.toImageFrame(), batchPerPartition = 1)
+
+      val prediction = output.transform(new FeatureTransformer {
```
Comment (jason-dai) [inline zoo/src/main/scala/com/intel/analytics/zoo/examples/tfnet/Predict.scala]: this is too complex for a demo. maybe just sth like:
```
output.rdd.map(_.predict).collect().print
```

**Comment by yangw1234** [comment]
pr validation pass

**Pull request merged**